### PR TITLE
Improve marketplace mesg.yml definition

### PR DESCRIPTION
**Pull request by NicolasMahe**
Better to check commit by commit.

Modifs:
- remove empty `name: ""` and `description: ""` (will add them back with content later).
- Use definition of `manifestServiceData` in `preparePublishServiceVersion`
- Add a ~~basic~~ full definition of service definition

### CODE REVIEW

**Comment by antho1404** [comment]
Any reason why the "configuration" is not part of the mesg.yml or is it just something you forgot ?
configuration has been extracted from dependencies

**Comment by NicolasMahe** [comment]
> Any reason why the "configuration" is not part of the mesg.yml or is it just something you forgot ?
> configuration has been extracted from dependencies

Nice catch. This PR has been done before the extraction of configuration ;)

File: systemservices/marketplace/mesg.yml [hunk 1]
```diff
@@ -11,129 +11,107 @@ configuration:
...
+                          hashVersion:
+                            type: String
+                          deployment:
+                            type: Object
+                            object:
+                              type:
+                                type: String
+                              source:
+                                type: String
+                          definition:
```
Comment (ilgooz) [inline systemservices/marketplace/mesg.yml]: I don't think we should add a "partial" validation for _definition_. 
to me, it's an all-or-nothing thing.

Comment (NicolasMahe) [reply systemservices/marketplace/mesg.yml]: Done

File: systemservices/marketplace/mesg.yml [hunk 2]
```diff
@@ -72,17 +72,99 @@ tasks:
...
+                                      type:
+                                        type: String
+                                      optional:
+                                        type: Boolean
+                                        optional: true
+                                      repeated:
+                                        type: Boolean
+                                        optional: true
+                                      object:
+                                        type: Object
```
Comment (ilgooz) [inline systemservices/marketplace/mesg.yml]: we could have typed Parameter here as well if we implemented this feature https://forum.mesg.com/t/service-definition-support-recursive-types/260/6

**Pull request merged**